src/database/operations/generators/generateCreateOrUpdateItemOperation.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Generates a function that will invoke a specific query for data insertion
+ *
+ * @template T - The type of the item being created.
+ * @template I - The type of object to be passed for data insertion.
+ *
+ * @param {string} queryName - The name of the query to execute.
+ * @param {Object} savedAttributes - A list of the attribute names which should be converted to query parameters.
+ *
+ * @returns {Function} A function that takes query parameters, limit, and offset, and returns a promise that resolves to a bundle of entries and total count.
+ */
+
+import { db as defaultDb } from '../../db';
+import {
+	getIsAdministratorFromAuthContext,
+	getKeycloakUserIdFromAuthContext,
+} from '../../../types';
+import type { AuthContext, JsonResultSet } from '../../../types';
+
+// This may seem silly but simply doing `keyof T` will not account for the possibility of `T` being a union type.
+type KeysOfUnion<T> = T extends T ? keyof T : never;
```

Comment:
I don't understand but I am not sure I want to either.

Reply:
I tried to update the comment to be more useful (but I agree it's better to just close your eyes sometimes when it comes to TypeScript)